libsignal-protocol/src/keys/identity_key_pair.rs

```diff
@@ -30,17 +33,36 @@ impl IdentityKeyPair {
         }
     }
 
-    pub fn public(&self) -> Result<PublicKey, Error> {
+    pub fn public(&self) -> PublicKey {
         unsafe {
             let raw = sys::ratchet_identity_key_pair_get_public(
                 self.raw.as_const_ptr(),
             );
             assert!(!raw.is_null());
-            Ok(PublicKey {
+            PublicKey {
                 raw: Raw::copied_from(raw),
-            })
+            }
+        }
+    }
+
+    pub fn private(&self) -> PrivateKey {
+        unsafe {
+            let raw = sys::ratchet_identity_key_pair_get_private(
+                self.raw.as_const_ptr(),
+            );
+            assert!(!raw.is_null());
+            PrivateKey {
+                raw: Raw::copied_from(raw),
+            }
         }
     }
 }
 
+impl Drop for IdentityKeyPair {
+    fn drop(&mut self) {
+        unsafe {
+            sys::ratchet_identity_key_pair_destroy(self.raw.as_ptr() as _)
```

Comment:
i'm not sure if we had to impl `Drop` too ?

Reply:
and it was a wrong dissension, we don't need to impl `Drop`, we will get a double-free problem xD

Reply:
Correct. Our `Raw<T>` smart pointer makes sure all objects which use signal's reference counting will be free'd at the correct time. 

Before #8 I was writing the `Drop` impls by hand and eventually gave up due to all the memory leaks and double-frees 😆